get_data_meteorological_multiworker.py

```diff
@@ -0,0 +1,249 @@
+
+                stations_resp = requests.get(request_uri_parameterized, headers=head)
+
+                if stations_resp.status_code != 200:
+                    logger.error(f"Error with request: {request_uri_parameterized}")
+                    logger.error(stations_resp)
+                    # We DONT want to kill the whole process
+                    #raise ValueError(f"Error obtaining {entity_desc} list")
+
+                current_results = stations_resp.json()
+
+                # Only process non-empty requests
+                if "metadata" not in current_results:
+
+                    no_results = True
+                    logger.debug(f"\tPulling for station {stn_id} for {year_current}: no results")
+                else:
+
+                    result_count = current_results["metadata"]["resultset"]["count"]
+
+                    current_batch_results = current_results["results"]
+                    list_obtained.extend(current_batch_results)
+
+                    logger.debug(f"\tPulling for station {stn_id} for {year_current}: {len(list_obtained)} total results")
+
+        logger.info(f"Pulling for station {stn_id}, readings from {year_start}-{year_end}, location: {stn_name}, completed with {len(list_obtained)} readings")
+
+        # Write CSV to disk
+        logger.debug(f"Writing completed pull for station {stn_id} to {out_file_csv}")
+        with open(out_file_csv, 'w') as output_file:
+            dict_writer = csv.DictWriter(output_file, fieldnames=field_list, delimiter="|")
+            dict_writer.writeheader()
+            dict_writer.writerows(list_obtained)
+
+    except Exception as err:
+        logger.error(f"ERROR: Exception encountered on pull for station {stn_id}")
+        logger.error(err)
+        traceback.print_exc()
+
+    cdo_tokens.put(license_key)
+
+    return list_obtained
+
+def get_climate_timeseries(stn_id):
+
+    request_uri = "https://www.ncdc.noaa.gov/cdo-web/api/v2/data"
+    data_fields = ["date", "datatype", "station", "attributes", "value"]
+
+    if not stn_id.startswith("GHCND"):
+        logger.warn(f"Skipping pull for non-GHCND station {stn_id}")
+        return
+
+    year_min = STATION_LOOKUP[stn_id]["year_min"]
+    year_max = STATION_LOOKUP[stn_id]["year_max"]
+
+    get_timeseries_batched(entity_desc="Daily Summaries",
+                           datasetid="GHCND",
+                           stn_id=stn_id,
+                           year_start=year_min,
+                           year_end=year_max,
+                           request_uri=request_uri,
+                           out_file_json=f"data_extracts_multiworker/{stn_id.replace(':','_')}.json",
+                           out_file_csv=f"data_extracts_multiworker/{stn_id.replace(':','_')}.csv",
+                           field_list=data_fields,
+                           extra_params=None)
+
+def get_list_batched(entity_desc, request_uri, out_file_json, out_file_csv, field_list, extra_params=None):
```

Comment:
can `extra_params` be `**kwargs`?

Reply:
If you want to go down the `extra_params`, I would suggest we let the end user pass in a dict to be rendered by the `params` parameter of the `GET` request. HOWEVER, I would suggest against doing something like `extra_params={}`, because of reasons laid out [here](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument).

You can either make the change to `**kwargs` or have `extra_params = None` AND

```
if extra_params is None:
    extra_params = {}
```